src/components/Query.ts

```diff
@@ -119,13 +94,54 @@ export class ViewEngine extends Engine {
 
 export class Queries {
   public static async buckets(connection: InfluxDBConnection): Promise<TableResult> {
-    if (connection.version == InfluxConnectionVersion.V1) {
+    if (connection.version === InfluxConnectionVersion.V1) {
       return await this.bucketsV1(connection);
     } else {
       return await this.bucketsV2(connection);
     }
   }
 
+  public static async measurements(connection: InfluxDBConnection, bucket: string): Promise<TableResult> {
+    if (connection.version === InfluxConnectionVersion.V1) {
+      return await this.measurementsV1(connection);
+    } else {
+      return await this.measurementsV2(connection, bucket);
+    }
+  }
+
+  public static async tagKeys(connection: InfluxDBConnection, bucket: string, measurement: string): Promise<TableResult> {
+    if (connection.version === InfluxConnectionVersion.V1) {
+      return this.tagKeysV1(connection, measurement);
+    } else {
+      return this.tagKeysV2(connection, bucket, measurement);
+    }
+  }
+
+  private static async tagKeysV1(connection: InfluxDBConnection, measurement: string): Promise<TableResult> {
+    const query = `show tag keys from ${measurement}`;
+    return await APIRequest.queryV1(connection, query);
+  }
+
+  private static async tagKeysV2(connection: InfluxDBConnection, bucket: string, measurement: string): Promise<TableResult> {
+    const query = `
+      import "influxdata/influxdb/v1"
+      v1.measurementTagKeys(bucket:"${bucket}", measurement: "${measurement}")`;
+
+    return await APIRequest.queryV2(connection, query);
+  }
+
+  private static async measurementsV1(connection: InfluxDBConnection): Promise<TableResult> {
+    const query = "show measurements";
+    return await APIRequest.queryV1(connection, query);
```

Comment:
bucket name is missing here, should be `APIRequest.queryV1(connection, query);`, v1 query requires bucket name

Reply:
v1query has a default empty string if the argument is not provided

Reply:
for `show databases;` can have an empty string, but `show measurements` will return nothing, if it's an empty string